### ISSUE TITLE
Problem playing sound after packaged electron app

### ISSUE THREAD

**Issue by DivadOger** [atom/electron#3965]
Hey,
I'm building an app on electron, currently only for OSX, that uses an external library (in c++) via Node Addon, that streams an audio file. This works well when running the app via "npm start" and via "electron music_app" or even if ran via "music_app.app/Contents/MacOS/Electron" after packaged. The problem is I can't get the sound working if i just do "open nmusic_app.app", but everything else runs fine.

I have really no clue about what's going on, is this any kind of known issue? I can't seem to find any information anywhere.

Thanks in advance.

**Comment by KeitIG** [comment]
Did you include the module in your build ? Does the chromium console output something ?

**Comment by DivadOger** [comment]
Do you mean the c++ node addon? It has to be included because if I run the app invoking the "music_app.app/Contents/MacOS/Electron" script directly everything goes fine. 
As for the second part of your question, there is no console if I double click on the "music_app.app" icon. (If there's a way to get a console log on this case, I'm all ears)
If I run via console as I mentioned earlier I do get an Audio Unit log:
```
AudioUnitGraph 0x1E96000:
  Member Nodes:
	node 1: 'aufc' 'conv' 'appl', instance 0x81e96069 O I
	node 2: 'aumx' 'mcmx' 'appl', instance 0x81e9606a O I
	node 3: 'auou' 'def ' 'appl', instance 0x81e9606b O I
  Connections:
	node   1 bus   0 => node   2 bus   0  [ 2 ch,  44100 Hz, 'lpcm' (0x00000029) 32-bit little-endian float, deinterleaved]
	node   2 bus   0 => node   3 bus   0  [ 2 ch,  44100 Hz, 'lpcm' (0x00000029) 32-bit little-endian float, deinterleaved]
  Input Callbacks:
	{0x11706ea30, 0x7ff83a0a6698} => node   1 bus   0  [2 ch, 44100 Hz]
  CurrentState:
	mLastUpdateError=0, eventsToProcess=F, isInitialized=T, isRunning=F
```

**Comment by KeitIG** [comment]
in your `main.js` file (the one where you create your `BrowserWindow`, you can add a `mainWindow.openDevTools()` to open the  console. Try to rebuild your .app with that instruction, and see what the console output when running your app.

**Comment by DivadOger** [comment]
Ah, that I already know of. The thing is my c++ module is loaded and everything goes fine! I can even call the addon functions directly on the dev console. The "only" thing is that sound doesn't play.
I don't think that the problem lies with the module but maybe with some environment differences when the app is ran via .app or directly via the .app shell script. Is there any real difference? I thought double clicking .app just launched the shell script.

**Comment by KeitIG** [comment]
And it should. Maybe I should let someone with electron knowledge answers. Usually, I try to avoid non-js or non-native modules in my apps. Take a look at the HTML5 web audio API, but it's another story :)

**Comment by DivadOger** [comment]
Thank you for your attention, and have a happy 2016 btw! 
I finally managed to fix the issue! What was happening is that the c++ library tries to cache some files in a temp directory, that I need to set before starting any song, but I was just setting it by passing something like ```('./temp-dir')```. So the thing is, if I ran from shell script or "npm start" it creates the dir right on actual directory, but if i ran by double clicking .app , the OSX doesn't know how to "resolve" that ```('./temp-dir')``` path and cannot create the temp directory. 
So in order to fix it, i used ```app.getPath('temp')```, from Electron, to set up that temp directory. Gladly it works fine!
